frontend/app.js

```diff
@@ -0,0 +1,81 @@
+// Home screen component
+const HomeComponent = {
+  render: (params) => homeScreen(params)
+}
+
+// Edit Schema screen component
+const SchemaComponent = {
+  render: (params) => schemaReport(params)
+}
+
+// Instructions Component
+const InstructionsComponent = {
+  render: (params) => renderInstructionsHtml(params)
+}
+
+// Error component (for any unrecognized path)
+const ErrorComponent = {
+  render: () => {
+    return `
+      <section>
+        <h1>Error</h1>
+      </section>
+    `;
+  }
+}
+
+// Pre defined routes 
+const routes = [
+  { path: '/', component: HomeComponent, },
+  { path: '/schema-report-connect-to-db', component: SchemaComponent, },
+  { path: '/schema-report-load-db-dump', component: SchemaComponent, },
+  { path: '/schema-report-import-db', component: SchemaComponent, },
+  { path: '/schema-report-resume-session', component: SchemaComponent, },
+  { path: '/instructions', component: InstructionsComponent, }
+];
+
+// function to fetch browser url
+const parseLocation = () => location.hash.slice(1).toLowerCase() || '/';
+
+// function to find component based on browser url
+const findComponentByPath = (path, routes) => routes.find(r => r.path.match(new RegExp(`^\\${path}$`, 'gm'))) || undefined;
+
+// function to render the html based on path
+const router = () => {
+  const path = parseLocation();
+  const { component = ErrorComponent } = findComponentByPath(path, routes) || {};
+
+  if (path == '/schema-report-connect-to-db' && event.type == 'hashchange') {
```

Comment:
Comparison path == '/schema-report-connect-to-db' may cause unexpected type coercion.

Solution: use === instead of == here.

Reply:
Agreed.. Replaced == with === where ever needed